### DISCUSSION TITLE
Allow to change path to remix.config.js

### DISCUSSION THREAD

**Discussion by suchy** [remix-run/remix#2020] [Proposals]
Hi,

it would be useful to be able to keep remix.config.js in a different directory than the root one.

In my current project remix isn't in the root directory. I'd like to be able to colocate config with the remix app.

Example:

```sh
cross-env NODE_ENV=development remix dev --config ./src/web/remix.config.js
```

Part of my current structure:

<img width="514" alt="Screenshot 2022-02-18 at 09 52 38" src="https://user-images.githubusercontent.com/255720/154650134-77f8cd9a-d720-4768-af68-5901dd598d95.png">

What do you think?

**Comment by sergiodxa** [comment]
I like this idea, I like to move my config files to a config folder and remix.config.js doesn't let me do that, I would love to be able to move it there

**Comment by suchy** [comment]
Wow, I proposed that almost a year ago :D I'd like to provide PR for that - for the anniversary 🎊 🎆 

My idea is to add a `--config-path` flag to `routes`, `build`, `watch`, and `dev` commands, then pass it to [readConfig](https://github.com/remix-run/remix/blob/ebfa6458af3f1e4ccf137200da05ac8f31df5a50/packages/remix-dev/config.ts#L306) function.

I already started playing with that and it works. 

As I understand, following the contribution guidelines, the team needs to accept that proposal, before I create PR.

**Comment by suchy** [comment]
After postcss and tailwind support release, I think it should be a path not to the config file but to the config directory - as right now there might be multiple config files
